### PR TITLE
Ants brain extraction

### PR DESCRIPTION
**Pull request by sgiavasis**

### CODE REVIEW

File: CPAC/pipeline/cpac_pipeline.py [hunk 1]
```diff
@@ -418,7 +418,7 @@ def prep_workflow(sub_dict, c, run, pipeline_timing_info=None,
 
         else:
 
-            if not any(o in c.skullstrip_option for o in ["AFNI", "BET"]):
+            if not any(o in c.skullstrip_option for o in ["AFNI", "BET", "antsBrainExtraction"]):
```
Comment (sgiavasis) [inline CPAC/pipeline/cpac_pipeline.py]: We should resolve naming first.

File: CPAC/resources/configs/pipeline_config_template.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# CPAC Pipeline Configuration YAML file
+ # CPAC Pipeline Configuration YAML file
```
Comment (sgiavasis) [inline CPAC/resources/configs/pipeline_config_template.yml]: Accidental space

File: CPAC/utils/interfaces/ants.py [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+"""
+Nipype interfaces for ANTs commands
+"""
+
+# This file is adapted from fMRIPrep
+#  more info: https://github.com/poldracklab/niworkflows/blob/master/niworkflows/interfaces/ants.py 
+#               https://fmriprep.readthedocs.io/ 
+
```
Comment (sgiavasis) [inline CPAC/utils/interfaces/ants.py]: Would suggest:

> This functionality is adapted from poldracklab/niworkflows:
>     https://github.com/poldracklab/niworkflows/blob/master/niworkflows/interfaces/ants.py
>     https://fmriprep.readthedocs.io/
>     https://poldracklab.stanford.edu/
> We are temporarily maintaining our own copy for more granular control.

File: CPAC/utils/interfaces/ants.py [hunk 4]
```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+"""
```
Comment (sgiavasis) [inline CPAC/utils/interfaces/ants.py]: Let's make sure to prune any unused functions.

File: CPAC/utils/interfaces/fixes.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+
+#  This file is adapted from fMRIPrep
+#  more info: https://github.com/poldracklab/niworkflows/blob/master/niworkflows/interfaces/fixes.py   
+#               https://fmriprep.readthedocs.io/ 
+
```
Comment (sgiavasis) [inline CPAC/utils/interfaces/fixes.py]: Same as above: credit and pruning.

File: CPAC/utils/misc.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+"""
+Miscellaneous utilities
+"""
+
```
Comment (sgiavasis) [inline CPAC/utils/misc.py]: Same as above: credit and pruning.

File: dev/docker_data/default_pipeline.yml [hunk 7]
```diff
@@ -110,6 +110,14 @@ reGenerateOutputs :  False
...
 skullstrip_option :  ['AFNI']
 
+# Template to be used during antsBrainExtraction.
+# It is not necessary to change this path unless you intend to use a non-standard template.
+# antsBrainExtraction Brain extraction template
+antsBrainExtraction_template_path : /cpac_templates/T_template0.nii.gz
+# antsBrainExtraction probability mask
+antsBrainExtraction_mask_path : /cpac_templates/T_template0_BrainCerebellumProbabilityMask.nii.gz
+# antsBrainExtraction registration mask 
+antsBrainExtraction_regmask_path : /cpac_templates/T_template0_BrainCerebellumRegistrationMask.nii.gz
```
Comment (sgiavasis) [inline dev/docker_data/default_pipeline.yml]: Maybe create a subdirectory called "OASIS_Template"

File: requirements.txt [hunk 8]
```diff
@@ -16,12 +16,13 @@ pandas==0.23.4
 patsy==0.5.0
 prov==1.5.0
 psutil==5.4.6
-pygraphviz==1.3.1
+# pygraphviz==1.3.1
 python-dateutil==2.7.3
```
Comment (sgiavasis) [inline requirements.txt]: ?

File: CPAC/utils/interfaces/utils.py [hunk 9]
```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+
+#  This file is adapted from fMRIPrep
+#  more info: https://github.com/poldracklab/niworkflows/blob/master/niworkflows/interfaces/utils.py  
+#               https://fmriprep.readthedocs.io/ 
+
+"""
```
Comment (sgiavasis) [inline CPAC/utils/interfaces/utils.py]: Same as above: credit and pruning.

File: CPAC/anat_preproc/ants.py [hunk 10]
```diff
@@ -0,0 +1,642 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+"""
+Nipype translation of ANTs workflows
+------------------------------------
+"""
+#  This file is adapted from fMRIPrep
+#  more info: https://github.com/poldracklab/niworkflows/blob/master/niworkflows/anat/ants.py 
+#               https://fmriprep.readthedocs.io/ 
+
```
Comment (sgiavasis) [inline CPAC/anat_preproc/ants.py]: See comments in other files regarding credit and pruning.

File: CPAC/anat_preproc/anat_preproc.py [hunk 11]
```diff
@@ -274,25 +296,62 @@ def create_anat_preproc(method='afni', already_skullstripped=False,
...
+                            outputnode, 'brain')
+
+        elif method == 'antsBrainExtraction': 
+            # Skull-stripping using antsBrainExtraction  
+            anat_skullstrip_ants = init_brain_extraction_wf(tpl_target_path=template_path,
+                                                            tpl_mask_path=mask_path,
+                                                            tpl_regmask_path=regmask_path,
+                                                            name='anat_skullstrip_ants')
+                                                            # use_float=True,
+                                                            # normalization_quality='precise',
```
Comment (sgiavasis) [inline CPAC/anat_preproc/anat_preproc.py]: Can remove these commented-out lines if not going to use.

File: CPAC/resources/configs/pipeline_config_template.yml [hunk 12]
```diff
@@ -109,9 +106,17 @@ runSymbolicLinks :  [1]
 already_skullstripped :  [0]
 
 
-# Choice of using AFNI or FSL-BET to perform SkullStripping
+# Choice of using AFNI or FSL-BET or antsBrainExtraction to perform SkullStripping
```
Comment (sgiavasis) [inline CPAC/resources/configs/pipeline_config_template.yml]: Let's rename to "niworkflows-ants"

File: dev/docker_data/default_pipeline.yml [hunk 13]
```diff
@@ -110,6 +110,14 @@ reGenerateOutputs :  False
 # Choice of using AFNI or FSL-BET to perform SkullStripping
```
Comment (sgiavasis) [inline dev/docker_data/default_pipeline.yml]: Add "niworkflows-ants" as third option in comment

File: CPAC/pipeline/cpac_pipeline.py (hunk 1 above)
Comment (sgiavasis) [reply CPAC/pipeline/cpac_pipeline.py]: Let's name it "niworkflows-ants"

File: CPAC/pipeline/cpac_pipeline.py [hunk 14]
```diff
@@ -501,6 +501,28 @@ def prep_workflow(sub_dict, c, run, pipeline_timing_info=None,
 
                 new_strat_list += [new_strat]
 
+            if "antsBrainExtraction" in c.skullstrip_option:
```
Comment (sgiavasis) [inline CPAC/pipeline/cpac_pipeline.py]: rename etc.

File: requirements.txt (hunk 8 above)
Comment (anibalsolon) [reply requirements.txt]: this is a cumbersome to install on Mac- I couldnt make it right on @HechengJin0 laptop.

we should move it to another pull request, but removing graphviz as a dependency would be nice. so we just move to another workflow plotting system.

Comment (sgiavasis) [reply requirements.txt]: Got it, thanks!

**Pull request merged**